Mage.Sets/src/mage/sets/ModernHorizons3.java

```diff
@@ -317,4 +326,159 @@ private ModernHorizons3() {
         cards.add(new SetCardInfo("Wumpus Aberration", 176, Rarity.UNCOMMON, mage.cards.w.WumpusAberration.class));
         cards.add(new SetCardInfo("Wurmcoil Larva", 112, Rarity.UNCOMMON, mage.cards.w.WurmcoilLarva.class));
     }
+
+    @Override
+    public List<Card> tryBooster() {
+        // TODO: make part of this more generic, this is the second try at a play booster generation so we try to see what can be shared.
+        // TODO: Commanders should appear sometimes as wild cards, add them to generation once they are implemented.
+        // source https://magic.wizards.com/en/news/feature/collecting-modern-horizons-3
+
+        // We start by deciding the various slots.
+        // some slots have guarantee rarity
+        int rareOrMythic = 1;
+        int newToModern = 1;
+        int uncommon = 3;
+        int common = 5; // 6 common slots, one being sometimes replaced by a secret guest
+
+        // Land Slot: 1/2 chance for a basic, 1/2 chance for a common
+        int basicLand = 0;
+        {
+            if (RandomUtil.nextDouble() <= 0.5) {
+                basicLand++;
+            } else {
+                common++;
+            }
+        }
+
+        // 1 slot is 1/64 chance to be spg, and 1/5 - 1/64 to be otp, 4/5 to be common
+        int spg = 0;
+        {
+            double rollSpg = RandomUtil.nextDouble();
+            if (rollSpg <= 1.0 / 64.0) { // know probability of 1/64 to be spg
+                spg++;
+            } else {
+                common++;
+            }
+        }
+
+        // 1 slot is a wildcard:
+        // 41.7% chance to be among 80 common
+        // 41.7% chance to be among 101 uncommons
+        // 7.8% chance to be among rare/mythic (with weight 2:1 individually)
+        // TODO: only that part has been generated (up to 91.2%).
+        //       miss some new to modern (not all) that have retro frames, (4.2%),
+        //       and the 8 Commander Mythic Rares (4.2% in total).
+        //       some variants (0.4%), and full art snow covered wastes (<0.1%)
```

Comment:
Looks overdeveloped.

Booster generation in xmage implemented by two ways:
* Random generation process with special settings likes ratio/num/etc. It's shared between all boosters and uses approximate data about it. There are additional methods to refine special slots and other set specific cards;
* Collation generation process. It's use a real life cards production. Required full data about set;

No needs to rewrite existing random generator for each new set.

Reply:
> Random generation process with special settings likes ratio/num/etc. It's shared between all boosters and uses approximate data about it. There are additional methods to refine special slots and other set specific cards;

Good luck with that. What can be in a booster and in which quantity has changed a lot since MKM. Booster generation without the extra cards/slots will just not be used by anyone.

Reply:
If current built-in booster generation incompatible with Play Boosters then create new issue about it (what's miss or wrong). Maybe it require some refactor (e.g. extract new generation as new booster class and use it for all new sets).

Reply:
There may be ways to make logic common as we get more examples of Play Booster generation. But it's not overdeveloped here - logic is too complex for existing xmage random generation due to sheer number of special slots and collation (I do not think it would be a good idea to add all sorts of extra params for it). It's maybe possible to use BoosterCollator for it, but without print runs (which only are available if a third party puts in the effort to research and publish), we would need to carefully add color balance functionality. And lots of careful code is necessary to support BoosterCollator, so I'm not even convinced it would be cleaner than what Susucr has here.

There's nothing "wrong" with existing logic. Just WoTC increasing complexity. For now, I think this is fine. There are todos and comments in the code here, although there's not any clear action to take at this point, we could create an issue with a brief summary.

Susucr and I will of course continue to seek opportunities for improvement, same as everything else.

Reply:
There is a [detailed description of Play Booster](https://mtg.fandom.com/wiki/Play_Booster). Other booster type also have same articles. Looks like wizards change booster structure once per few years. It was introduced for Murders at Karlov Manor, but that's set don't have boosters at all in xmage.

I'm surprised why xmage don't implement different booster types same way as wizard's research team does. It can be better approach instead single bloated generator.

![shot_240606_095344](https://github.com/magefree/mage/assets/8344157/36945414-868b-4d93-ab57-161002bad551)